pkg/controller/batch/batch_controller.go

```diff
@@ -0,0 +1,355 @@
+	}
+
+	if spec.ConfigMap != nil && spec.Config != nil {
+		return reconcile.Result{},
+			r.UpdatePhase(v2.BatchFailed, fmt.Errorf("At most one of ['Spec.config', 'spec.ConfigMap'] must be configured"))
+	}
+	return reconcile.Result{}, r.UpdatePhase(v2.BatchInitializing, nil)
+}
+
+func (r *batchResource) initializeResources() (reconcile.Result, error) {
+	batch := r.instance
+	spec := batch.Spec
+	// Ensure the Infinispan cluster exists
+	infinispan := &v1.Infinispan{}
+	if result, err := kube.LookupResource(spec.Cluster, batch.Namespace, infinispan, r.client, log); result != nil {
+		return *result, err
+	}
+
+	if err := infinispan.EnsureClusterStability(); err != nil {
+		log.Info(fmt.Sprintf("Infinispan '%s' not ready: %s", spec.Cluster, err.Error()))
+		return reconcile.Result{RequeueAfter: consts.DefaultWaitOnCluster}, nil
+	}
+
+	if spec.ConfigMap == nil {
+		// Create configMap
+		configMap := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      batch.Name,
+				Namespace: batch.Namespace,
+			},
+		}
+		_, err := controllerutil.CreateOrUpdate(ctx, r.client, configMap, func() error {
+			configMap.Data = map[string]string{BatchFilename: *spec.Config}
+			return controllerutil.SetControllerReference(batch, configMap, r.scheme)
+		})
+
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("Unable to create ConfigMap '%s': %w", configMap.Name, err)
+		}
+
+		updated, err := r.update(func() error {
+			batch.Spec.ConfigMap = &batch.Name
+			return controllerutil.SetControllerReference(infinispan, batch, r.scheme)
+		})
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("Unable to update Batch .Spec: %w", err)
+		}
+		if updated {
+			return reconcile.Result{}, nil
+		}
+	}
+
+	// We update the phase separately to the spec as the status update is ignored when in the update mutate function
+	return reconcile.Result{}, r.UpdatePhase(v2.BatchInitialized, nil)
+}
+
+func (r *batchResource) execute() (reconcile.Result, error) {
+	batch := r.instance
+	infinispan := &v1.Infinispan{}
+	if result, err := kube.LookupResource(batch.Spec.Cluster, batch.Namespace, infinispan, r.client, log); result != nil {
+		return *result, err
+	}
+
+	url, err := connectionUrl(infinispan)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	cliArgs := fmt.Sprintf("--connect '%s' --trustall --file '%s/%s'", url, BatchVolumeRoot, BatchFilename)
+
+	labels := batchLabels(batch.Name)
+	infinispan.AddLabelsForPods(labels)
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      batch.Name,
+			Namespace: batch.Namespace,
+		},
+		Spec: batchv1.JobSpec{
+			BackoffLimit: pointer.Int32Ptr(0),
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Name:    batch.Name,
+						Image:   infinispan.ImageName(),
```

Comment:
@ryanemerson is it possible to use small CLI image instead of full server image for the Job?

Reply:
Thanks, make sense.